### PR TITLE
Implement UserRepository.currentUserId

### PR DESCRIPTION
**Pull request by hideya**
Implement UserRepository.currentUserId

The id is set by the login and cleared by the logout methods.
The id is preserved in NSUserDefaults across app restarts.

/to @bajtos could you please review?

### CODE REVIEW

**Comment by bajtos** [comment]
The build seems to be failing due to misconfiguration of the build slave ([full log](http://ci.strongloop.com/job/loopback-sdk-ios.pr/3/console)):

```
/bin/sh -xe /var/folders/kw/jk6rg86170l80qn0w5g36jmr0000gq/T/hudson6030942266181753250.sh
  xcodebuild -sdk iphoneos
  -derivedDataPath /Users/jenkins/build/workspace/loopback-sdk-ios.pr/derived 
  -project LoopBack.xcodeproj -scheme LoopBackFramework -configuration Debug
  ONLY_ACTIVE_ARCH=NO build

xcode-select: Error: No Xcode is selected. Use xcode-select -switch <path-to-xcode>,
  or see the xcode-select manpage (man xcode-select) for further information.

Build step 'Execute shell' marked build as failure
```

@rmg could you please check and fix the problem?

**Comment by bajtos** [comment]
FWIW, LGTM. @raymondfeng could you PTAL too?

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
CI is working again :-)

**Comment by hideya** [comment]
Added the following changes:
- Add findCurrentUser and cachedCurrentUser to LBUserRepository
- Make some of the LBUserTests tighter and independently executable.

File: LoopBack/LBUser.m [hunk 1]
```diff
@@ -91,9 +117,35 @@ - (void)logoutWithSuccess:(LBUserLogoutSuccessBlock)success
                      success:^(id value) {
                          LBRESTAdapter* adapter = (LBRESTAdapter*)self.adapter;
                          adapter.accessToken = nil;
+                         self.currentUserId = nil;
```
Comment (bajtos) [inline LoopBack/LBUser.m]: You should set `self.cachedCurrentUser = nil` as well. Please add a test (or extend an existing test) to cover this contract too.

Comment (bajtos) [inline LoopBack/LBUser.m]: Oh, I see, you are clearing `cachedCurrentUser` whenever `currentUserId` is changed. Ignore my previous comment then.

File: LoopBackTests/LBUserTests.m [hunk 2]
```diff
@@ -53,61 +62,187 @@ @implementation LBUserTests
...
-- (void)testCreate {
+- (void)testCreateSaveRemove {
     ASYNC_TEST_START
-    Customer __block *user = (Customer*)[self.repository createUserWithEmail:@"testUser@test.com"
-                                                                    password:@"test"];
-    [user saveWithSuccess:^{
+    double uid = [[NSDate date] timeIntervalSince1970];
+    NSString *userEmail = [NSString stringWithFormat:@"%f%@", uid, USER_EMAIL_DOMAIN];
+
+    Customer __block *customer = (Customer*)[self.repository createUserWithEmail:userEmail password:USER_PASSWORD];
```
Comment (bajtos) [inline LoopBackTests/LBUserTests.m]: It's better to keep the lines at 80 characters max, that way they fit in the diff windows on GitHub and the reviewer does not have to scroll the text horizontally.

**Comment by bajtos** [comment]
FWIW, LGTM.

@raymondfeng could you PTAL too?

**Comment by raymondfeng** [comment]
LGTM

**Comment by bajtos** [comment]
@hideya can you please rebase your patch on top of the latest master and clean up the commit history?

**Comment by hideya** [comment]
@bajtos, I've rebased and cleaned up the commit history.  Please take a look.

**Comment by bajtos** [comment]
Landed, thank you!

**Pull request merged**